### ISSUE TITLE
Test Scaleway servers

### ISSUE THREAD

**Issue by robnagler** [radiasoft/devops#70]
[scaleway.com](http://scaleway.com) will be coming out with high performance servers for low cost.

Test servers:

- [ ] single node
- [ ] multi-node
- [ ] file system
- [ ] move alpha-jupyter.radiasoft.org 
- [ ] move alpha.sirepo.com

**Comment by robnagler** [comment]
Turns out that the pricing was not bare metal. Scaleway's best bare metal config is:

* €23.99/Month
* 8 Dedicated x86 64bit Cores
* 32GB Memory
* 50GB SSD Disk
* 250GB Direct SSD Disk
* 1 Flexible public IPv4
* 800Mbit/s Internet bandwidth
* 5Gbit/s Internal bandwidth

This is still very good. Rackspace OnMetal is:

* OnMetal General Purpose Small	
* 32G
* 1x6
* 2x800GB hot-swappable SSDs configured in a RAID 1 mirror
* 10,000 Mb/s
* $725.00/mo

[Leaseway](https://www.leaseweb.com/bare-metal-server/configure/25081) offers:

* €189.00
* 2x Intel 12-Core Xeon E5-2650v4
* 128GB DDR4
* 4x480GB SSD

**Comment by robnagler** [comment]
The servers work, but there's a significant issue with the way disks work.

There have been issues with the way the disks work. The bare metal servers have two types of partitions: Local SSD (LSSD) and Direct SSD (DSSD). LSSD is actually a network disk [NBD](https://nbd.sourceforge.io/) (link doesn't work right now), which is not a local disk, despite the name. DSSD is a local disk.

C2M instances ($20/mo) only work with NBD disks. To use them with Docker, you need to partition the disk for the meta and data device-mapper partitions. The fdisk works, but the partition table is not read until a reboot. That's ok, but then when the machine boots, the /dev/nbd1p1 and /dev/nbdp2 partitions don't show up. I've been having a conversation with the tech people.

C2M instances come with one 250gb DSSD disk. The DSSD is special in that when you power off the instance, the disk goes away, forever. LSSD (NBD) disks are archived and restored when you power on the instance. This is distinct from terminating an instance, where you would expect disks to go away. LSSD is archived when you power off the instance so they can reuse the physical machine. DSSD are not archived when you power off. 

The archive of the LSSD takes some time (minute or so for 50gb) as does the restore. Not critical, but something to know.

At this point, I think things would work, but I'm still waiting on the fix to the nbd issue. I could create the /dev files manually, but it seems wrong.

**Comment by robnagler** [comment]
I'm giving up on Scaleway. It's super cheap, but it's taken me days to struggle against their network configuration. They don't support VPCs (AWS) or a backnet (Linode). You have to go their their masqueraded IPs. MPI does not like this at all.

@elventear suggested to use swarm, but this makes the firewall configuration (#83) much more complicated, because overlay networks require even more iptables configuration. It also isn't clear that all these layers of networking an proxying is going to help the performance. We don't need to prove the concept, because somebody else has [MPI running over overlay networks](https://github.com/carlochess/docker.openmpi).

Another issue with Scaleway is that you can't partition nbd disks, which is ok, but annoying so you have to allocate 50GB for docker metadata or use a directly partitioned disk.

The price is unbeatable though...
